File: indigo-backend/src/main/java/com/martinia/indigo/notification/domain/model/Notification.java
package com.martinia.indigo.notification.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification implements Serializable {

	private String id;
	private NotificationEnum type;
	private String user;
	private Date date;
	private boolean readUser;
	private boolean readAdmin;
	private StatusEnum status;
	private String message;
	private NotificationEpubFileUploadItem upload;
	private NotificationKindleItem kindle;

}

File: indigo-frontend/src/app/domain/notification.ts
import { NotificationEnum } from '../enums/notification.enum.';
import { NotificationKindle } from '../domain/notification.kindle';
import { NotificationUpload } from '../domain/notification.upload';
import { StatusEnum } from '../enums/status.enum';

export class Notification {

    constructor(
        public id?:string,
        public type?:NotificationEnum,
        public user?: string,
        public date?:string,
        public readUser?:boolean,
        public readAdmin?:boolean,
        public status?:StatusEnum,
        
        public kindle?:NotificationKindle,
        public upload?:NotificationUpload,

        public message?:string
        ) {
    }
}
